packages/sitecore-jss/src/personalize/graphql-personalize-service.ts

```diff
@@ -82,22 +86,25 @@ export class GraphQLPersonalizeService {
       language
     );
 
-    const data = await this.graphQLClient.request<PersonalizeQueryResult>(this.query, {
-      siteName: this.config.siteName,
-      itemPath,
-      language,
-    });
+    try {
+      const data = await this.graphQLClient.request<PersonalizeQueryResult>(this.query, {
+        siteName: this.config.siteName,
+        itemPath,
+        language,
+      });
+
+      const personalizeInfo = !data?.layout?.item
+        ? undefined
+        : {
+            // CDP expects content id format `<id>_<language>_<version>` (lowercase)
+            contentId: `${data.layout.item.id}_${language}_${data.layout.item.version}`.toLowerCase(),
+            segments: data.layout.item.personalization.variantIds,
+          };
 
-    if (!data?.layout?.item) {
-      // Possible if itemPath / language doesn't exist
+      return personalizeInfo;
+    } catch (error) {
```

Comment:
It means that we handle all errors, but it shouldn't happen, error should be thrown, Our purpose to return default value only in case of timeout